### DISCUSSION TITLE
C++ with WebWorker

### DISCUSSION THREAD

**Discussion by AlphaNecron** [TypeFox/monaco-languageclient#434] [Q&A]
I'm intending to create an editor for C++ online, but without a local LSP server. Is it possible to run a language server in the web browser with WebWorker?

**Comment by kaisalmen** [comment]
Hi @AlphaNecron it is possible to run a LSP server in a web worker. This is what done in the browser-lsp example: https://github.com/TypeFox/monaco-languageclient/tree/main/packages/examples/browser-lsp

But, the LSP code must be written in something that can be executed in the context of a web worker (JavaScript or WASM).

**Reply by AlphaNecron** [reply]
Do you know any LSP out there which is compatible with this? Python, C++, Clang, C#, etc..

**Reply by kaisalmen** [reply]
I haven't heard of any for the languages you listed.
Best overview of available language server implementations I know of is this:
https://microsoft.github.io/language-server-protocol/implementors/servers/

**Reply by CGNonofr** [reply]
btw, there is ways to compile c++ into wasm, and some file system implementations exist for the browser, so maybe it's possible

https://github.com/binji/wasm-clang managed to run the clang compiler in the browser

**Reply by AlphaNecron** [reply]
Oh thanks. I'll check it out.
